streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -146,7 +146,7 @@ public void process(final K key, final V1 value) {
 
                     outerJoinWindowStore.ifPresent(store -> {
                         // Delete the joined record from the non-joined outer window store
-                        store.put(KeyAndJoinSide.make(!isLeftSide, key), null, otherRecordTimestamp);
+                        store.put(KeyAndJoinSide.make(!isLeftSide, key, otherRecordTimestamp), null);
```

Comment:
Should you call `store.delete(KeyAndJoinSide.make(!isLeftSide, key, otherRecordTimestamp))` now?

Reply:
Same as above.